source/opt/instruction.h

```diff
@@ -93,6 +93,17 @@ class Instruction {
   Instruction(Instruction&&);
   Instruction& operator=(Instruction&&);
 
+  // Returns the position for this instruction or one of its operand.
+  // The position is encoded as a spv_postion_t struct, with its
+  // * line field to mean the position of this instruction in the module
+  //   (starting from 1, 0 means invalid position),
+  // * column field to mean the the position of operand in this instruction
```

Comment:
Another weird thing.  Why not use the OpLine instructions to indicate position.  Whose position is it really?  Some high level language source file position, or inside-the-binary-module position.  It's not really clear except by comments.  This data could be easily misused.  (Also, position inside the binary changes with code transforms, discussed elsewhere in this review.)